### PR TITLE
Update documentation for OTA design

### PR DESCRIPTION
**Pull request by divekarshubham**
*Issue #, if available:*

*Description of changes:*
* Moved `OTA_SignalEvent()` to `ota.h` 
* Updated documentation for OTA Design page
* Added dependency graph 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/doxygen/pages.dox [hunk 1]
```diff
@@ -1,9 +1,9 @@
 /**
 @mainpage
 @anchor ota
-@brief OTA (over-the-air) library.<br><br>
+@brief Over-the-air (OTA) library.<br><br>
```
Comment (nihaljamazon) [inline docs/doxygen/pages.dox:4]: Update the name as "AWS IoT Over-the-air Update Library (Release Candidate)"

File: docs/doxygen/pages.dox [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
+AWS IOT Over-the-air Update library enables you to manage the notification of a newly available update, download the update, and perform cryptographic verification of the firmware update. Using the Over-the-air (OTA) client library, you can logically separate firmware updates from the application running on your devices. The Over-the-air (OTA) client library can share a network connection with the application, saving memory in resource-constrained devices. In addition, the OTA library lets you define application-specific logic for testing, committing, or rolling back a firmware update. The library supports different application protocols like Message Queuing Telemetry Transport (MQTT) and Hypertext Transfer Protocol (HTTP) and provides various configuration options you can fine tune depending on network type and conditions.
```
Comment (nihaljamazon) [inline docs/doxygen/pages.dox:6]: IOT => IoT

Comment (divekarshubham) [reply docs/doxygen/pages.dox]: Addressed in https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/128/commits/85064419842169800cd88b04dee3c1248cf7c474

File: docs/doxygen/pages.dox (hunk 1 above)
Comment (divekarshubham) [reply docs/doxygen/pages.dox]: Addressed in https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/128/commits/85064419842169800cd88b04dee3c1248cf7c474

**Review by nihaljamazon — APPROVED** [review]

**Review by abhidixi11 — APPROVED** [review]

**Pull request merged**